components/calendar/calendar.js

```diff
@@ -696,100 +727,100 @@ class Calendar extends LocalizeCoreElement(RtlMixin(LitElement)) {
+					this._triggerMonthChangeAnimations(oldFocusDate < possibleFocusDate, true, Math.abs(numDaysChange) !== 1);
 				}
-			} else {
-				this._focusDateAddFocus();
-			}
-			await this.updateComplete;
-			this._isInitialFocusDate = true;
+				return;
+			} else return;
 		}
+		else this._focusDate = possibleFocusDate;
+		await this._showFocusDateMonth(oldFocusDate, Math.abs(numDaysChange) !== 1);
+	}
+
+	async _onNextMonthButtonClick() {
+		this._monthIncrease();
+		this._triggerMonthChangeAnimations(true);
+		await this._updateFocusDateOnChange();
+	}
+
+	async _onPrevMonthButtonClick() {
+		this._monthDecrease();
+		this._triggerMonthChangeAnimations(false);
+		await this._updateFocusDateOnChange();
 	}
 
-	_onNextMonthButtonClick() {
-		this._updateShownMonthIncrease();
-		this._updateFocusDateOnChange();
+	async _showFocusDateMonth(prevFocusDate, upDownAnimation) {
+		this._isInitialFocusDate = false;
+		const date = await this._getDateElement(this._focusDate);
+		if (!date) {
+			this._keyboardTriggeredMonthChange = true;
+			let increase;
+			if (prevFocusDate < this._focusDate) {
+				increase = true;
+				this._monthIncrease();
+			} else {
+				increase = false;
+				this._monthDecrease();
+			}
+			this._triggerMonthChangeAnimations(increase, true, upDownAnimation);
+		} else {
+			this._focusDateAddFocus();
+		}
+		await this.updateComplete;
+		this._isInitialFocusDate = true;
 	}
 
-	_onPrevMonthButtonClick() {
-		this._updateShownMonthDecrease();
-		this._updateFocusDateOnChange();
+	_triggerMonthChangeAnimations(increase, keyboardTriggered, transitionUpDown) {
+		this._monthNav = undefined;
+		increase = this.dir === 'rtl' ? !increase : increase;
+		if (!keyboardTriggered) this._isInitialFocusDate = true;
+		if (!reduceMotion) {
+			setTimeout(() => {
+				this._monthNav = `${increase ? 'next' : 'prev'}${transitionUpDown ? '-updown' : ''}`;
+			}, 100); // timeout for firefox
+		}
 	}
 
-	async _updateFocusDateDependentOnDisabled(possibleFocusDate, latestPossibleFocusDate) {
+	async _updateFocusDate(possibleFocusDate, latestPossibleFocusDate, allowDisabled) {
 		await this.updateComplete;
-		if (isDateInRange(possibleFocusDate, getDateFromISODate(this.minValue), getDateFromISODate(this.maxValue))) {
+		if (isDateInRange(possibleFocusDate, getDateFromISODate(this.minValue), getDateFromISODate(this.maxValue)) || allowDisabled) {
 			this._focusDate = possibleFocusDate;
+			return true;
 		} else if (this.shadowRoot.querySelector('.d2l-calendar-date:enabled')) {
 			const validDates = this.shadowRoot.querySelectorAll('.d2l-calendar-date:enabled');
 			const focusDate = validDates[latestPossibleFocusDate ? (validDates.length - 1) : 0].parentNode;
 			const year = focusDate.getAttribute('data-year');
 			const month = focusDate.getAttribute('data-month');
 			const date = focusDate.getAttribute('data-date');
 			this._focusDate = new Date(year, month, date);
+			return true;
 		} else {
-			this._focusDate = undefined;
+			return false;
```

Comment:
`_focusDate` should not be set to undefined (other than with the page up/down case, which I moved in there) as this causes problems when all the dates in a month are disabled and the user presses multiple buttons